package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -228,6 +239,29 @@ def __init__(self, universe, donors_sel=None, hydrogens_sel=None, acceptors_sel=
         self.donors_sel = donors_sel
         self.hydrogens_sel = hydrogens_sel
         self.acceptors_sel = acceptors_sel
+
+        # If hydrogen bonding groups are selected, then generate corresponding atom groups
+        if between is not None:
+
+            if type(between) is not list or len(between)==0:
```

Comment:
Use `isinstance()` instead of the `type() == ...` checks (also below).

Or do you really have to type check? Can’t you just work with anything that will allow you to do the iteration, e.g., a numpy array? Handle with try/except and duck typing is probably more pythonic.

Reply:
I've attempted to address this and will read more about duck typing. Thanks